lms/static/scripts/frontend_apps/components/ErrorDisplay.js

```diff
@@ -30,82 +30,114 @@ function toSentence(str) {
   }
+  return details;
+}
 
-  let supportEmailBody = '';
-  if (error.message) {
-    supportEmailBody += `\n\nError message: ${error.message}`;
-  }
-  if (details) {
-    supportEmailBody += `\n\nTechnical details:\n\n${details}`;
+/**
+ * @typedef ErrorDetailsProps
+ * @prop {ErrorLike} error
+ *
+ * Render pre-formatted JSON details of an error
+ *
+ * @param {ErrorDetailsProps} props
+ */
+function ErrorDetails({ error }) {
+  /** @param {Event} event */
+  const onDetailsToggle = event => {
+    const detailsEl = /** @type {HTMLDetailsElement} */ (event.target);
+    if (!detailsEl.open) {
+      return;
+    }
+    detailsEl.scrollIntoView({ behavior: 'smooth' });
+  };
+
+  const details = formatErrorDetails(error);
+  if (!details) {
+    return null;
   }
 
+  return (
+    <details className="ErrorDisplay__details" onToggle={onDetailsToggle}>
+      <summary className="ErrorDisplay__details-summary">Error Details</summary>
+      <pre className="ErrorDisplay__details-content">{details}</pre>
+    </details>
+  );
+}
+
+/**
+ * @typedef ErrorDisplayProps
+ * @prop {string|null} [description] -
+ *   A short message explaining the error and its human-facing relevance.
+ *   This is typically a general message like
+ *   "There was a problem fetching this assignment". This description
+ *   always comes from (this) client app.
+ *
+ *   The presence of a `description` indicates that this `ErrorDisplay` is
+ *   responsible for explaining the error to the user in some fashion. Along
+ *   with this `description`, anything available in `error.message` is also
+ *   shown.
+ *
+ *   When `description` is absent, it indicates that the main explaining of
+ *   the error's relevance to the user is handled elsewhere, and no additional
+ *   error messaging is necessary. This is the case, for example, with some
+ *   of the well-explained error states in `OAuth2RedirectErrorApp` or
+ *   `LaunchErrorDialog`: these don't need additional error messaging that is,
+ *   for the most part, redundant or useless.
+ *
+ *   Available `error.details` and support/email instructions are always
+ *   shown regardless of the presence of this prop.
+ *
+ * @prop {ErrorLike} error - Error-like object containing further `details`
+ *   or `message` about this error state. The value of `details` and `message`
+ *   may come from a server response.
+ */
+
+/**
+ * Displays human-facing details of an error, including:
+ * - Explanation/message (if `description` is provided)
+ * - Instructions for contacting support and getting help
+ * - Stringified-JSON details of the error (if `error.details` is populated)
+ *
+ * @param {ErrorDisplayProps} props
+ */
+export default function ErrorDisplay({ description, error }) {
+  const details = formatErrorDetails(error);
+
   const supportLink = emailLink({
     address: 'support@hypothes.is',
     subject: 'Hypothesis LMS support',
-    body: supportEmailBody,
+    body: `
+Error message: ${error?.message || 'N/A'}
```

Comment:
If the `error.message` is undefined, should we try to use the description?
```suggestion
Error message: ${error?.message || description || 'N/A'}
```

Reply:
Great idea! I'm going to make it a separate line, however, so it's never ambiguous what its source is.